### PR TITLE
🐶 Add Structs samples and lab

### PR DESCRIPTION
**Pull request by juliaogris**
### Changes proposed in this PR:

* Hack highlight.js to not highlight text/bash as `go` (revert when #11  is merged) 
* Add struct slides and lab
* Scaffold remaining samples in sample/README.md and slide links
* Add third party libs used at ANZ slide

### CODE REVIEW

File: content/gotraining.slide [hunk 1]
```diff
@@ -507,23 +507,29 @@ As part of all labs
...
+- [[https://goplay.space/#An9Um8oSShm][Interfaces]]
 
 
-* Lab 5
+* Lab 5 - CRUD puppy with interface
 
-TODO
+- Create an executable go program in directory `5_puppy`
+- Implement a `Puppy` struct containing `ID`, `Breed`, `Colour`, `Value`.
+- Create `Storer` interface with [[https://en.wikipedia.org/wiki/Create,_read,_update_and_delete][CRUD] methods for `Puppy`
```
Comment (camscale) [inline content/gotraining.slide]: Broken link formatting - missing trailing close bracket (`]`).

File: content/gotraining.slide [hunk 2]
```diff
@@ -619,8 +625,14 @@ Use `gofmt` or `goimport` as part of your editor/IDE setup
...
-- TODO
-
+- logrus
+- pq
+- redigo
+- mgo
+- goracle
+- jsonschema
+- go-jose
+- ...
```
Comment (camscale) [inline content/gotraining.slide]: When you're done with this list, order it alphabetically (perhaps), add a brief description, and link to the project pages/repos.

If you want, I'll sort it out while you work on the slides and samples.

File: samples/structs.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// Structs
+package main
+
+import "fmt"
+
+// This `person` struct type has `name` and `good` fields.
```
Comment (camscale) [inline samples/structs.go]: Does it? Does it really? no good. it's malicious.

Fixed: https://goplay.space/#eoBLEu8lRvl

File: content/gotraining.slide [hunk 4]
```diff
@@ -507,23 +507,29 @@ As part of all labs
 
 * Data structures and interfaces
 
-- Pointers
-- Structs
-- Methods
-- Interfaces
+- [[https://goplay.space/#MYeoiSdP4tp][Pointers]]
+- [[https://goplay.space/#lHNpzvoFabD][Structs]]
```
Comment (camscale) [inline content/gotraining.slide]: New malicious link: https://goplay.space/#eoBLEu8lRvl

File: content/gotraining.slide [hunk 5]
```diff
@@ -507,23 +507,29 @@ As part of all labs
...
 
+.image _img/crud_puppy.jpg
 
 * Errors
 
-- Errors
-- Panic
-- Recover
-- Defer
+- [[][Errors]]
```
Comment (camscale) [inline content/gotraining.slide:540]: oh, you're sneaky. flipping through the slides seeing all these links I thought you'd been a very busy beaver. But it's good to have no errors.

File: content/gotraining.slide [hunk 6]
```diff
@@ -507,23 +507,29 @@ As part of all labs
 
 * Data structures and interfaces
 
-- Pointers
-- Structs
-- Methods
-- Interfaces
+- [[https://goplay.space/#MYeoiSdP4tp][Pointers]]
+- [[https://goplay.space/#lHNpzvoFabD][Structs]]
+- [[https://goplay.space/#mg9ScasagOC][Methods]]
```
Comment (camscale) [inline content/gotraining.slide]: New link: https://goplay.space/#OkA4buOlTe1

Remove import parens

File: samples/methods.go [hunk 7]
```diff
@@ -0,0 +1,35 @@
+// Methods
+package main
+
+import (
+	"fmt"
+)
```
Comment (camscale) [inline samples/methods.go]: Remove parens for consistency with other slides.

File: samples/methods.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
+// Methods
```
Comment (camscale) [inline samples/methods.go:1]: New version: https://goplay.space/#OkA4buOlTe1

File: samples/interfaces.go [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+import "fmt"
+
+type scaler interface {
+	scale(float64)
+}
+
+func main() {
+	var s scaler
+	s = &rect{2, 3}
```
Comment (camscale) [inline samples/interfaces.go:12]: speaker notes: ask the class, "why didn't we do `s := &rect{2, 3}` instead of these two lines?"

**Review by camscale — COMMENTED** [review]
a few small fixups

File: content/gotraining.slide (hunk 1 above)
Comment (juliaogris) [reply content/gotraining.slide]: Done.

File: samples/structs.go (hunk 3 above)
Comment (juliaogris) [reply samples/structs.go]: :)
Done.

File: content/gotraining.slide (hunk 2 above)
Comment (juliaogris) [reply content/gotraining.slide]: There is a somewhat implied order in how frequently/commonly they get used.
Acceptable?
The list is complete-ish for now.

File: content/gotraining.slide (hunk 5 above)
Comment (juliaogris) [reply content/gotraining.slide:540]: I had busy beaver ambitions... but alas, just human.

File: content/gotraining.slide (hunk 6 above)
Comment (juliaogris) [reply content/gotraining.slide]: again. shame on me. sorry!
Done.

File: samples/methods.go (hunk 7 above)
Comment (juliaogris) [reply samples/methods.go]: Done.

File: samples/interfaces.go (hunk 9 above)
Comment (juliaogris) [reply samples/interfaces.go:12]: Great idea!
Will do.

File: content/gotraining.slide (hunk 2 above)
Comment (juliaogris) [reply content/gotraining.slide]: and yes, if you feel like linking and collating the info, i'd be even more appreciative than i already am. wait a minute - is that possible?

**Comment by juliaogris** [comment]
PTAL @camh-anz (force push), dropped the hljs hack in favor of rebase onto @pentaphobe 's fixup.
Also, haven't linked and described the third party libs yet.
Not sure about sorting them alphabetically - I listed them in subjective order of frequency used.

File: content/gotraining.slide (hunk 6 above)
Comment (camscale) [reply content/gotraining.slide]: this link has not been updated.

File: content/gotraining.slide (hunk 2 above)
Comment (camscale) [reply content/gotraining.slide]: i don't mind either way. i think the list is on the boundary of wanting to be sorted, but if it is in the order you want it, keep it that way.

Comment (juliaogris) [reply content/gotraining.slide]: ok, cool.... adding the links now myself if that's ok.

**Comment by juliaogris** [comment]
@camh-anz - PTAL. I snuck in another exercise because I felt the CRUD might be too big a starting step - hope that's ok.

**Review by camscale — APPROVED** [review]
LGTM

3P ordering is better now it is grouped.

Extra lab is also good - implementing existing interfaces is so common it makes a good starting point.

**Pull request merged**